go/enclave/db/db.go

```diff
@@ -7,27 +7,26 @@ import (
 
 	"github.com/obscuronet/go-obscuro/go/enclave/db/sql"
 
-	"github.com/ethereum/go-ethereum/core/rawdb"
-	"github.com/ethereum/go-ethereum/ethdb"
 	"github.com/obscuronet/go-obscuro/go/config"
 )
 
 // CreateDBFromConfig creates an appropriate ethdb.Database instance based on your config
-func CreateDBFromConfig(cfg config.EnclaveConfig, logger gethlog.Logger) (ethdb.Database, error) {
+func CreateDBFromConfig(cfg config.EnclaveConfig, logger gethlog.Logger) (sql.EnclaveDB, error) {
 	if err := validateDBConf(cfg); err != nil {
 		return nil, err
 	}
 	if cfg.UseInMemoryDB {
 		logger.Info("UseInMemoryDB flag is true, data will not be persisted. Creating in-memory database...")
-		return getInMemDB()
+		// this creates an in memory sqllite db
+		return sql.CreateTemporarySQLiteDB("file:"+cfg.HostID.Hex()+"?mode=memory&cache=shared", true, logger)
```

Comment:
Feels a bit duplicative to be defining the db as in-mem with the filename and then passing that flag in too. I wonder how it would come out to make a separate method like `sql.CreateInMemorySQLiteDB(cfg.HostID.Hex(), logger)` that builds the `file:` and `mode=` bits too.

Or maybe just rename the `inMem` bool parameter to `overwriteExisting` so it feels more general?

Reply:
unfortunately, the sqlite "memory" setting doesn't work very well. I keep getting "table is locked" errors. Something to do with parallel usage. 
I'll use the temp file option.